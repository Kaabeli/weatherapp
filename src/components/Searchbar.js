import React from 'react';


const Searchbar = (props) => {

        return (
            <div>
                <h3>City Weather Search</h3>
                <form onSubmit={props.searchWeather}>
                <label>Search for City: </label>
                <input onChange={props.eventHandler} id="city" name="city" />
                <button type="submit">Search</button>
                </form>
            </div>
        );
    }

export default Searchbar;
