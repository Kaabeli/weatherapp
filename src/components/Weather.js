import React, { Component } from 'react';
import Searchbar from './Searchbar';
import { isNullOrUndefined } from 'util';



class Weather extends Component {

    constructor() {
        super();
        this.state = {
            temperature: null,
            weatherDes: '',
            icon: '',
            city: '',
            showcity: '',
            searchData: [],
            isData: '',
        }
    }

    eventHandler = (e) => {
        this.setState({
            city: e.target.value
        });
    }


    searchWeather = (e) => {
        e.preventDefault();
        fetch("http://api.openweathermap.org/data/2.5/weather?q=" + this.state.city + "&units=metric&APPID=b452af841ef38883145b7302033d8de9")
            .then((response) => response.json())
            .then((responseData) => {
                if (responseData.cod === "404") {
                    this.setState({
                        isData: isNullOrUndefined,
                    })
                } else {
                    this.setState({
                        showcity: this.state.city,
                        temperature: responseData.main.temp,
                        weatherDes: responseData.weather[0].description,
                        icon: "http://openweathermap.org/img/w/" + responseData.weather[0].icon + ".png",
                        isData: true,
                    });
                }
            })
    }

    addCity() {
        const addSearchedCity = {
            showcity: this.state.city,
            temperature: this.state.temperature,
            weatherDes: this.state.weatherDes
        }
        this.setState({
            searchData: [
                ...this.state.searchData, addSearchedCity
            ]
        })
    }

    renderIsData() {
        if (this.state.isData === true)
            return (
                <div>
                    <h4> Weather of {this.state.showcity}</h4>
                    <div>Temperature: {this.state.temperature} Celsius</div>
                    <div>Weather: {this.state.weatherDes}</div>
                    <div><img src={this.state.icon} alt="icon" /></div>
                    <label>Click to save:</label>
                    <button type="submit" onClick={() => this.addCity()}>Save</button>
                    <br />
                </div >
            );
    }

    renderError() {
        if (this.state.isData === isNullOrUndefined)
            return (
                <div>
                    <h6>City not found!</h6>
                    <br />
                </div>
            );
    }

    renderCityData() {
        return (
            <tbody>
                {this.state.searchData.map((data, index) => {
                    return (
                        <tr key={index}>
                            <td>{data.showcity}</td>
                            <td>{data.temperature}</td>
                            <td>{data.weatherDes}</td>
                            <td><button onClick={() => this.removeCity(index)}>Remove</button></td>
                        </tr>)
                })}
            </tbody>
        );
    }


    removeCity(index) {
        const searchData = [...this.state.searchData];
        searchData.splice(index, 1);
        this.setState({
            searchData
        });
    }


    render() {
        return (
            <div className="container">

                <Searchbar
                    city={this.state.city}
                    searchWeather={this.searchWeather.bind(this)}
                    eventHandler={this.eventHandler.bind(this)} />
                <br />
                {this.renderError()}
                {this.renderIsData()}
                <table className="table">
                    <thead>
                        <tr>
                            <th scope="col">City</th>
                            <th scope="col">Temperature</th>
                            <th scope="col">Weather</th>
                            <th scope="col">Remove</th>
                        </tr>
                    </thead>
                    {this.renderCityData()}
                </table>


            </div>
        );
    }
}

export default Weather;
