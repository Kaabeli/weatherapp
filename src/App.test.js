import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Weather from './components/Weather';
import Searchbar from './components/Searchbar';
import renderer from 'react-test-renderer';

it('App renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Weather renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Weather />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Searchbar renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Searchbar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Searchbar renders correctly', () => {
  const tree = renderer.create(<Searchbar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Weather renders correctly', () => {
  const tree = renderer.create(<Weather />).toJSON();
  expect(tree).toMatchSnapshot();
});