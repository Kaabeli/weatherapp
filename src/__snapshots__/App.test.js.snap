// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div>
      <h3>
        City Weather Search
      </h3>
      <form
        onSubmit={[Function]}
      >
        <label>
          Search for City: 
        </label>
        <input
          id="city"
          name="city"
          onChange={[Function]}
        />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <br />
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            City
          </th>
          <th
            scope="col"
          >
            Temperature
          </th>
          <th
            scope="col"
          >
            Weather
          </th>
          <th
            scope="col"
          >
            Remove
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Searchbar renders correctly 1`] = `
<div>
  <h3>
    City Weather Search
  </h3>
  <form>
    <label>
      Search for City: 
    </label>
    <input
      id="city"
      name="city"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`Weather renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <h3>
      City Weather Search
    </h3>
    <form
      onSubmit={[Function]}
    >
      <label>
        Search for City: 
      </label>
      <input
        id="city"
        name="city"
        onChange={[Function]}
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <br />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          City
        </th>
        <th
          scope="col"
        >
          Temperature
        </th>
        <th
          scope="col"
        >
          Weather
        </th>
        <th
          scope="col"
        >
          Remove
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
